src/main/java/com/coreos/jetcd/lease/KeepAlive.java

```diff
@@ -1,65 +1,72 @@
 package com.coreos.jetcd.lease;
 
-import com.coreos.jetcd.Lease.LeaseHandler;
-import com.coreos.jetcd.api.LeaseGrantResponse;
+import com.coreos.jetcd.LeaseImpl.KeepAliveListenerImpl;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
 
 /**
  * The KeepAlive hold the keepAlive information for lease.
  */
 public class KeepAlive {
 
-  private final long leaseID;
-
-  private long ttl;
-
-  private LeaseGrantResponse leaseGrantResponse;
-
   private long deadLine;
 
   private long nextKeepAlive;
 
-  private LeaseHandler leaseHandler;
+  // ownerLock protects owner map.
+  private final Object ownerLock;
 
-  public KeepAlive(long leaseID) {
-    this(leaseID, null);
-  }
+  private Map<Long, KeepAlive> owner;
```

Comment:
the concurrentHashMap is passed in from leaseImpl through its constructor. So it will hold a reference to a concurrent hashmap.